### ISSUE TITLE
0 splash screen detected

### ISSUE THREAD

**Issue by greenozon** [kiraio-moe/USSR#29]
I've tried the latest build from [here](https://github.com/kiraio-moe/USSR/issues/28#issuecomment-2359820327) (windows)
and it says it found 0 results

is it expected?
```
Unity Splash Screen Remover v1.1.8

USSR is a tool to easily remove Unity splash screen. USSR didn't directly "hack" the Unity Editor, but the generated
build.

Before using USSR, make sure you have set the splash screen Draw Mode in Player Settings to All Sequential and backup
the target file below! For more information, visit USSR GitHub repo: https://github.com/kiraio-moe/USSR

HOW TO USE
Select the action below, find and select one of these files in your game data:
globalgamemanagers | data.unity3d | *.data | *.data.br | *.data.gz | *.data.unityweb


Opening File Picker...
( INFO ) Selected file: C:\Tmp\2\Build\xxx\globalgamemanagers
( INFO ) Loading class types package: C:\Users\user\Downloads\USSR-net8.0-win-x64-latest\classdata.tpk...
( INFO ) Loading asset file: C:\Tmp\2\Build\xxx\globalgamemanagers.temp...
( INFO ) Loading asset class types database...
( INFO ) Unity Version: 2022.3.30f1
( INFO ) Start removing Unity splash screen...
( INFO ) There's 0 splash screen detected:
( WARN ) Nothing to do.
( INFO ) Backup globalgamemanagers as C:\Tmp\2\Build\xxx\globalgamemanagers.bak...
( INFO ) Writing changes to C:\Tmp\2\Build\xxx\globalgamemanagers...
( INFO ) Cleaning up temporary files...

Press any key to continue...

```

**Comment by greenozon** [comment]
Also even saying 0 splash detected in fact it did modify the globalgamemanagers  file and created .bak one
is that expected?

**Comment by kiraio-moe** [comment]
> I've tried the latest build from [here](https://github.com/kiraio-moe/USSR/issues/28#issuecomment-2359820327) (windows) and it says it found 0 results
> 
> is it expected?

Yes, some Unity versions aren't able to read and it's leading to it.

**Comment by kiraio-moe** [comment]
> Also even saying 0 splash detected in fact it did modify the globalgamemanagers file and created .bak one is that expected?

I forgot to check to not write the file if it's not modified.

**Comment by greenozon** [comment]
is it possible to add newer versions? or it needs  some kind of R.E. in order to support it...

**Comment by kiraio-moe** [comment]
> is it possible to add newer versions? or it needs some kind of R.E. in order to support it...

An easier way you can do is just updating the `classdata.tpk` from https://github.com/AssetRipper/Tpk.

**Comment by greenozon** [comment]
which one do I need to update - 
Type Tree Tpk
or
Engine Assets Tpk

should I rename  uncompressed.tpk  into classdata.tpk?

**Comment by kiraio-moe** [comment]
> which one do I need to update - Type Tree Tpk or Engine Assets Tpk

Type Tree.

> should I rename uncompressed.tpk into classdata.tpk?

Absolutely.

**Comment by Sidiusz** [comment]
seems that unity 6000 not supported even with the new brotli tpk :(

```
( ERR! ) Error when loading class types package! Unable to read beyond the end of the stream.
Unity Splash Screen Remover v1.1.8
```

**Comment by kiraio-moe** [comment]
> seems that unity 6000 not supported even with the new brotli tpk :(
> 
> ```
> ( ERR! ) Error when loading class types package! Unable to read beyond the end of the stream.
> Unity Splash Screen Remover v1.1.8
> ```

Hi, Unity 6 allows you to disable Unity splash screen entirely. This tools only intended to work with Unity up to 2024.

**Comment by Sidiusz** [comment]
> > seems that unity 6000 not supported even with the new brotli tpk :(
> > ```
> > ( ERR! ) Error when loading class types package! Unable to read beyond the end of the stream.
> > Unity Splash Screen Remover v1.1.8
> > ```
> 
> Hi, Unity 6 allows you to disable Unity splash screen entirely. This tools only intended to work with Unity up to 2024.

Yes, I understand, but I don't need a disable, I need a complete cut of the logo assets from the build

**Comment by kiraio-moe** [comment]
> Yes, I understand, but I don't need a disable, I need a complete cut of the logo assets from the build

Why do you need that? I don't see the use case for that.

**Comment by Sidiusz** [comment]
> > Yes, I understand, but I don't need a disable, I need a complete cut of the logo assets from the build
> 
> Why do you need that? I don't see the use case for that.

because it's a useless logo and a lot of other things in the Unity build that take up free space
